pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -267,9 +269,7 @@ impl Timeline {
             // shard but an old generation, or they belonged to another shard.  This also implicitly
             // guarantees that the layer is persistent in remote storage (as only remote persistent
             // layers are carried across shard splits, any local-only layer would be in the current generation)
-            if layer.metadata().generation == self.generation
-                && layer.metadata().shard.shard_count == self.shard_identity.count
-            {
+            if layer.metadata().generation == self.generation {
```

Comment:
nit: can you update the comment above as well?

Reply:
done.